### ISSUE TITLE
RVM Not installing old Ruby versions

### ISSUE THREAD

**Issue by mrsalmanahmad** [rvm/rvm#5217]
## Description
I am stuck on this for two days. I am unable to install any ruby version using rvm. I tried to uninstall brew and install it again but the issue is still there. Please don't mind i am very new to ruby on rails. thanks for the help. 

Here i have attached all the console errors and steps i performed. 
## Steps to reproduce

➜  ~ rvm | head -n 1                                               
Ruby enVironment Manager 1.29.12 (latest) (c) 2009-2020 Michal Papis, Piotr Kuczynski, Wayne E. Seguin
➜  ~ rvm lisr known
Unrecognized command line argument: lisr
Run `rvm help` to see usage information
➜  ~ rvm list known
➜  ~ rvm | head -n 1
Ruby enVironment Manager 1.29.12 (latest) (c) 2009-2020 Michal Papis, Piotr Kuczynski, Wayne E. Seguin
➜  ~   curl -sSL https://get.rvm.io | bash -s stable

Downloading https://github.com/rvm/rvm/archive/1.29.12.tar.gz
Downloading https://github.com/rvm/rvm/releases/download/1.29.12/1.29.12.tar.gz.asc
Found PGP signature at: 'https://github.com/rvm/rvm/releases/download/1.29.12/1.29.12.tar.gz.asc',
but no GPG software exists to validate it, skipping.
Upgrading the RVM installation in /Users/macbookpro/.rvm/
    RVM PATH line found in /Users/macbookpro/.mkshrc /Users/macbookpro/.profile /Users/macbookpro/.bashrc /Users/macbookpro/.zshrc.
    RVM sourcing line found in /Users/macbookpro/.profile /Users/macbookpro/.bash_profile /Users/macbookpro/.zlogin.
Upgrade of RVM in /Users/macbookpro/.rvm/ is complete.

Thanks for installing RVM 🙏
Please consider donating to our open collective to help us maintain RVM.

👉  Donate: https://opencollective.com/rvm/donate

➜  ~ rvm use 2.1 --default
Required ruby-2.1.10 is not installed.
To install do: 'rvm install "ruby-2.1.10"'
➜  ~ rmv install "ruby-2.1.10"
zsh: command not found: rmv
➜  ~ rvm install "ruby-2.1.10"                      
Searching for binary rubies, this might take some time.
No binary rubies available for: osx/12.3/x86_64/ruby-2.1.10.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
Checking requirements for osx.
Installing requirements for osx.
Updating system - please wait
Installing required packages: autoconf, automake, libtool, pkg-config, coreutils, libyaml, libksba, readline, zlib, openssl - please wait
There were package installation errors, make sure to read the log.

Try `brew tap --repair` and make sure `brew doctor` looks reasonable.

Check Homebrew requirements https://docs.brew.sh/Installation
Error running 'requirements_osx_brew_libs_install autoconf automake libtool pkg-config coreutils libyaml libksba readline zlib openssl',
please read /Users/macbookpro/.rvm/log/1652608512_ruby-2.1.10/package_install_autoconf_automake_libtool_pkg-config_coreutils_libyaml_libksba_readline_zlib_openssl.log
Requirements installation failed with status: 1.
➜  ~ 

 1. I have installed brew correctly
 2. Installed RVM attached its version above 
 3. Still not installing any ruby

## Expected behavior

It should work as brew and rvm is showing correct outputs. But when i am installing ruby its just not working. 

➜  ~ rvm info

system:

  system:
    uname:        "Darwin macbooks-MacBook-Pro.local 21.4.0 Darwin Kernel Version 21.4.0: Fri Mar 18 00:45:05 PDT 2022; root:xnu-8020.101.4~15/RELEASE_X86_64 x86_64"
    name:         "OSX"
    version:      "12.3"
    architecture: "x86_64"
    bash:         "/bin/bash => GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin21)"
    zsh:          "/bin/zsh => zsh 5.8 (x86_64-apple-darwin21.0)"
    remote_path:  "osx/12.3/x86_64"
    xcode:        "13.3.1"

  rvm:
    version:      "1.29.12 (latest)"
    updated:      "12 minutes 29 seconds ago"
    path:         "/Users/macbookpro/.rvm"
    autolibs:     "[4] Allow RVM to use package manager if found, install missing dependencies, install package manager (only OS X)."

  homes:
    gem:          "not set"
    ruby:         "not set"

  binaries:
    ruby:         "/usr/bin/ruby"
    irb:          "/usr/bin/irb"
    gem:          "/usr/bin/gem"
    rake:         "/usr/bin/rake"

  environment:
    PATH:         "/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin:/Users/macbookpro/.rvm/bin"
    GEM_HOME:     ""
    GEM_PATH:     ""
    MY_RUBY_HOME: ""
    IRBRC:        ""
    RUBYOPT:      ""
    gemset:       ""

**Comment by BrianHawley** [comment]
Looks like you're trying to install Ruby 2.1, which last came out in 2016, is end-of-life, and requires openssl 1.0. I would not be surprised if it won't install on modern macOS because of the old openssl requirement.

Pardon the question, but if you're new to Ruby on Rails, why are you trying to install such an old Ruby version? "I am unable to install any ruby version using rvm." - have you tried installing the latest version of Ruby and working your way back from there?

You can start by getting the latest rvm bug fixes like this:
```
rvm get master
```

Then you can try installing the latest Ruby with a command like:
```
rvm install ruby-3.1
```
Or even just:
```
rvm install ruby
```

If you need to use Ruby 2, you can install 2.7 with:
```
rvm install ruby-2
```

**Comment by mrsalmanahmad** [comment]
No i have a different error. I think my brew is not installed properly

Last login: Tue May 17 00:31:59 on ttys000
➜  ~ rvm get master
Downloading https://get.rvm.io
No GPG software exists to validate rvm-installer, skipping.
Downloading https://github.com/rvm/rvm/archive/master.tar.gz
Upgrading the RVM installation in /Users/macbookpro/.rvm/
    RVM PATH line found in /Users/macbookpro/.mkshrc /Users/macbookpro/.profile /Users/macbookpro/.bashrc /Users/macbookpro/.zshrc.
    RVM sourcing line found in /Users/macbookpro/.profile /Users/macbookpro/.bash_profile /Users/macbookpro/.zlogin.
Upgrade of RVM in /Users/macbookpro/.rvm/ is complete.

Thanks for installing RVM 🙏
Please consider donating to our open collective to help us maintain RVM.

👉  Donate: https://opencollective.com/rvm/donate


RVM reloaded!
➜  ~ rvm install ruby-3.1

Searching for binary rubies, this might take some time.
No binary rubies available for: osx/12.3/x86_64/ruby-3.1.2.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
Checking requirements for osx.
Installing requirements for osx.
Updating system - please wait
Failed to update Homebrew, follow instructions at

    https://docs.brew.sh/Common-Issues

and make sure `brew update` works before continuing.
Error running 'requirements_osx_brew_update_system ruby-3.1.2',
please read /Users/macbookpro/.rvm/log/1652730222_ruby-3.1.2/update_system.log
Requirements installation failed with status: 1.
➜  ~

**Comment by mathieujobin** [comment]
close issue?

**Comment by bigos** [comment]
Please do not close the issue for the sake of closing it and do something about it. For many years, rvm has been an easy-to-use solution for managing rubies. People who live in the real world have to maintain legacy Ruby applications and surprises like this are not what we need. As a temporary solution, I was forced to move to rbenv. I hope rbenv is only a temporary workaround.

**Comment by pkuczynski** [comment]
We are very small team and its very hard to manage all combinations of old ruby and newer OS. We are open for PRs from community if anybody has time to discover why older ruby does not build on newer OS...

**Comment by bigos** [comment]
Maybe there is no solution, and for this case, I will have to maintain an old VM to support the old code. All the hacks that promise the fix to deprecated ssl library either do not work or are too scary to try.

**Comment by mathieujobin** [comment]
Mac OSX 12.x is still a lot of pain to get many rubies to compile...

**Comment by conathan** [comment]
I threw together a patch to have Ruby 1.8.7 and Ruby 2.0.0 compile against OpenSSL 1.1
#5164 -  Patch Ruby 1.8.7 to compile against openssl 1.1.x
#5165 -  Patch Ruby 2.0.0 to compile against openssl 1.1.x

(I wanted ruby to compile against the system versions of openssl in brew/linux at the time, vs having a standalone OpenSSL 1.0 to compile against)

Can be applied by using rvm install ... --patches (and depending on the system, some changes to rvm may be needed to disable trying to install libssl1.0-dev  or other similarly named packages on other os's).



Ruby 2.4.x was the first version to support OpenSSL 1.1.x (And Ruby 3.1 is the first to support OpenSSL 3.0)

**Comment by pkuczynski** [comment]
@conathan we would be happy to merge #5163 and #5165 if they would be provided as PR's, and not github issues... Would you like to work on that?

**Comment by conathan** [comment]
@pkuczynski 

I wasn't expecting them to be accepted honestly,  but thought if I posted them then if someone really wants them they can use them.  (and in the off chance they were accepted, more eyes would certainly not hurt)


Sure, I'll throw something together on this when I get the chance, but I have only tested this with Ruby 1.8.7 (latest) and Ruby 2.0.  (also have a Ruby 1.9.3 version, but we never ended up using that ourselves [but can include it]).  

Also I personally don't fully trust that the patches are perfect,  (They compile, and we are using this inhouse ourselves), but it is a large patch.
